src/Lecture2.hs

```diff
@@ -1,170 +1,215 @@
-{- |
-Module                  : Lecture2
-Copyright               : (c) 2021-2022 Haskell Beginners 2022 Course
-SPDX-License-Identifier : MPL-2.0
-Maintainer              : Haskell Beginners 2022 Course <haskell.beginners2022@gmail.com>
-Stability               : Stable
-Portability             : Portable
-
-Exercises for the Lecture 2 of the Haskell Beginners course.
-
-As in the previous section, implement functions and provide type
-signatures. If the type signature is not already written, write the
-most polymorphic type signature you can.
-
-Unlike exercises to Lecture 1, this module also contains more
-challenging exercises. You don't need to solve them to finish the
-course but you can if you like challenges :)
--}
-
+-- |
+-- Module                  : Lecture2
+-- Copyright               : (c) 2021-2022 Haskell Beginners 2022 Course
+-- SPDX-License-Identifier : MPL-2.0
+-- Maintainer              : Haskell Beginners 2022 Course <haskell.beginners2022@gmail.com>
+-- Stability               : Stable
+-- Portability             : Portable
+--
+-- Exercises for the Lecture 2 of the Haskell Beginners course.
+--
+-- As in the previous section, implement functions and provide type
+-- signatures. If the type signature is not already written, write the
+-- most polymorphic type signature you can.
+--
+-- Unlike exercises to Lecture 1, this module also contains more
+-- challenging exercises. You don't need to solve them to finish the
+-- course but you can if you like challenges :)
 module Lecture2
-    ( -- * Normal
-      lazyProduct
-    , duplicate
-    , removeAt
-    , evenLists
-    , dropSpaces
-
-    , Knight (..)
-    , dragonFight
-
-      -- * Hard
-    , isIncreasing
-    , merge
-    , mergeSort
-
-    , Expr (..)
-    , Variables
-    , EvalError (..)
-    , eval
-    , constantFolding
-    ) where
+  ( -- * Normal
+    lazyProduct,
+    duplicate,
+    removeAt,
+    evenLists,
+    dropSpaces,
+    Knight (..),
+    dragonFight,
+
+    -- * Hard
+    isIncreasing,
+    merge,
+    mergeSort,
+    Expr (..),
+    Variables,
+    EvalError (..),
+    eval,
+    constantFolding,
+  )
+where
 
 -- VVV If you need to import libraries, do it after this line ... VVV
 
--- ^^^ and before this line. Otherwise the test suite might fail  ^^^
+import Data.Char
+import Data.Fixed (E0)
```

Comment:
I think this import was probably imported by an accident, and you don't actually use it here, so you can safely delete it 🙂